Dockerfile

```diff
@@ -4,10 +4,12 @@ ARG RUNTIME_IMAGE
 # Use the build image variant (eg heroku-18-build) which include headers/build tools.
 FROM $BUILD_IMAGE as builder
 
-ARG REDIS_VERSION
+# Using a prefixed variable name, to prevent the compile from accidentally relying on it
+# instead of remembering to load in /env/REDIS_VERSION as required by the buildpack API.
+ARG TEST_REDIS_VERSION
```

Comment:
I think we also need to update `test.sh` to pass this new name. At the moment we haven't. The result is that the `TEST_REDIS_VERSION` from the env matrix is being ignored, so we are installing the default redis version in every build, and so the build is still green. I wonder if we can make the build fail if a typo like this happens again?

Reply:
(For anyone reading this - these are notes to myself given @edmorley is away, so not expecting a response. But feel free to comment if you have something to say!)

Reply:
We now check the redis version that ended up being installed, which covers this.